Kunena/Kunena-Forum
# Add unit tests and fix bugs and issues found during unit tests

Add unit tests
- Add tests for CategoryUserHelper, CategoryUser and TopicUserHelper
- Improve KunenaForumCategoryUserHelper::getCategories()
- Improve KunenaForumTopicUserHelper::getTopics()
- Test Posting and moving topics
- Add collection of new framework tests
- Restructure tests

Fix bugs and issues found during unit tests
- Fix KunenaForumTopicUser::load()
- Fix KunenaForumTopicUser::delete()
- General code improvements
- Rename KunenaForumCategory::getUser() into getUserInfo()
- Make KunenaForumCategory::load(null) to reload current object
- Make KunenaForumCategory::load/delete() to register instance
- Add new function KunenaForumCategory::purge()
- Delete also keywords in KunenaForumCategory::delete()
- Fix KunenaForumCategory::update() on moved topic or same time
- Add function KunenaForumCategoryHelper::register()
- Add option 'categories' inKunenaForumCategoryHelper::recount()
- Fix KunenaForumCategoryHelper::recount(): posts missing 
- Fix KunenaForumMessageHelper::loadLocation(), order is by time
- Don't use same id when saving topic in KunenaForumMessage::save()
- Use unique posting time in KunenaForumMessage::check()
- Modify checks in TableKunenaMessages::check()
- Add instances lookup by topic in KunenaForumTopicUserHelper
- KunenaForumTopicUserHelper::move(), update objects as well
- Fix wrong table prefix in KunenaForumTopicUserHelper::merge()
- KunenaForumTopicUserHelper::merge(), update objects as well
- Fix deleting too many entries in KunenaForumTopicUserHelper::recount()
- Add function KunenaForumTopicUserHelper::reloadTopic()
- Fix KunenaForumTopicUser contructor: set always topic id
- Add function KunenaForumTopicUser::reset()
- KunenaForumTopicUser::load(): do not load if there's no topic id
- Make KunenaForumTopic::load(null) to reload current object
- Fix bugs in KunenaForumTopic::move() when moving individual messages
- Ensure unique posting time in KunenaForumTopic::move()
- Fix some bugs in KunenaForumTopic::save()
- Fix bugs in KunenaForumTopic::updatePostInfo(), update()

Add KunenaException class
